src/main/java/de/themoep/snap/forwarding/listener/ClientConnectListener.java

```diff
@@ -34,10 +34,12 @@ public ClientConnectListener(Snap snap) {
 
     @Subscribe(order = PostOrder.FIRST)
     public void on(PreLoginEvent event) {
-        ClientConnectEvent e = snap.getBungeeAdapter().getPluginManager().callEvent(new ClientConnectEvent(
+        ClientConnectEvent e = new ClientConnectEvent(
                 event.getConnection().getRemoteAddress(),
                 snap.getBungeeAdapter().getProxy().getListener()
-        ));
+        );
+        e.setCancelled(!event.getResult().isAllowed());
+        snap.getBungeeAdapter().getPluginManager().callEvent(e);
         if (e.isCancelled()) {
             event.setResult(PreLoginEvent.PreLoginComponentResult.denied(Component.text("ClientConnectEvent Cancelled")));
```

Comment:
While I know that it's PostOrder.FIRST, pre-cancelling this would override any result message set by another Velocity plugin if it happens to go before Snap. Maybe only updating the message if Velocity's PreLoginEvent didn't already have a message set would be good.